### PR TITLE
Add editor javascript docstrings

### PR DESCRIPTION
**Pull request by PhilippMatthes**
Closes: #302

### CODE REVIEW

File: assets/js/editor.js [hunk 1]
```diff
@@ -191,17 +316,36 @@ class ModalConfirmationForm extends Modal {
...
+     *
+     * @param {string} newTitle - The new title.
+     */
     changeTitle(newTitle) {
         this.elem.title = newTitle;
         this.elem.attr("data-original-title", newTitle)
     }
 
+    /**
+     * Shows the tooltip for a given amount of time.
```
Comment (martinmo) [inline assets/js/editor.js]: "... for *the* given *duration*."

File: assets/js/editor.js [hunk 2]
```diff
@@ -103,14 +155,34 @@ class ModalNotification extends Modal {
...
+     * @param {string} title - The title of the notification.
+     */
     constructor(title) {
         super(title, "Please try again later.");
     }
 }
 
 
+/**
+ * Represents a notification modal with custom body and title, which declares
```
Comment (martinmo) [inline assets/js/editor.js]: "...which *notifies* the user that a filename already exists."

File: assets/js/editor.js [hunk 3]
```diff
@@ -502,11 +818,25 @@ class TabBar {
 }
 
 
+/**
+ * Represents the code editor.
+ *
+ * Serves as a convenience wrapper for the base editor.
```
Comment (martinmo) [inline assets/js/editor.js]: * What does "base editor" refer to? The Ace.js editor? If so, say it directly.
* Add some high-level info about how the editor works and how it reuses Ace's features. For example: one Ace `Editor` object, a `Session`s for each tab (is that the way it works? I don't know yet).
* Add some high-level info about how this `Editor` class is supposed to be used.

**Comment by martinmo** [comment]
Btw, it looks to me that `completion` should have it's own class + documentation? Is that possible? (If so, in another PR, of course.)

**Comment by martinmo** [comment]
One more thing: the coding convention section of the contributor guide should be updated, to refer to JSDoc style comments.

File: assets/js/editor.js (hunk 3 above)
Comment (PhilippMatthes) [reply assets/js/editor.js]: I added an example (in a future commit) to explain how to use the editor and more high level information on how the editor wraps `ace.js`.

**Comment by PhilippMatthes** [comment]
> Btw, it looks to me that `completion` should have it's own class + documentation? Is that possible? (If so, in another PR, of course.)

Actually no, completion is just a function. Maybe we can address the inconsistent naming between `completion`, `success`, `closure`... in a future issue, but this is not critical.

**Comment by martinmo** [comment]
> > Btw, it looks to me that `completion` should have it's own class + documentation? Is that possible? (If so, in another PR, of course.)
> 
> Actually no, completion is just a function. Maybe we can address the inconsistent naming between `completion`, `success`, `closure`... in a future issue, but this is not critical.

Then I don't understand what `completion.{html,container,modal}` are.

**Comment by PhilippMatthes** [comment]
> > > Btw, it looks to me that `completion` should have it's own class + documentation? Is that possible? (If so, in another PR, of course.)
> > 
> > 
> > Actually no, completion is just a function. Maybe we can address the inconsistent naming between `completion`, `success`, `closure`... in a future issue, but this is not critical.
> 
> Then I don't understand what `completion.{html,container,modal}` are.

These are the arguments, which must be provided to the completion function ;)

**Comment by martinmo** [comment]
> These are the arguments, which must be provided to the completion function ;)

But the object property syntax is used, which is confusing ;)

**Comment by martinmo** [comment]
https://jsdoc.app/tags-callback.html

**Pull request merged**